cmd/otel-allocator/allocation/strategy.go

```diff
@@ -92,11 +92,22 @@ func Register(name string, provider AllocatorProvider) error {
 	return nil
 }
 
+func GetRegisteredAllocatorNames() []string {
+	var names []string
+	for s := range registry {
+		if len(s) > 0 {
```

Comment:
Curious in what case would len(s) == 0? Wondering if this is rare enough that it might be worthwhile to preallocate `names` with the size len(registry) on line 96.

Reply:
I actually found some cases in testing where there were empty strings in this list and figured this would be good defensive programming, the more i think of it, the more i think the problem was actually in how i was setting the names. I can remove this.